service/awsconfig/v4/key/key.go

```diff
@@ -175,24 +218,6 @@ func RoleName(customObject v1alpha1.AWSConfig, profileType string) string {
 	return fmt.Sprintf("%s-%s-%s", ClusterID(customObject), profileType, RoleNameTemplate)
 }
 
-// RootDir returns the path in the base directory until the
-// root elemant is found.
-func RootDir(baseDir, rootElement string) (string, error) {
-	items := strings.Split(baseDir, string(filepath.Separator))
-	rootIndex := -1
-	for i := len(items) - 1; i >= 0; i-- {
-		if items[i] == rootElement {
-			rootIndex = i
-			break
-		}
-	}
-	if rootIndex == -1 {
-		return "", microerror.Mask(notFoundError)
-	}
-
-	return "/" + filepath.Join(items[:(rootIndex+1)]...), nil
-}
-
```

Comment:
In v4 we do not need the file system magic anymore. Note that all the other versions still do it the old way. We have to cleanup certain things over time.

Reply:
We'll also be deleting the older resources eventually. I'm not sure it makes sense to update v2 and v3.

Reply:
I would just keep them as they are. We can move forward from here. Good enough for me.